File: 백준/Silver/18258. 큐 2/큐 2.cc
#include <iostream>
#include <queue>

using namespace std;

queue<int> q;

int main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    int N;
    cin >> N;

    for (int i = 0; i < N; ++i)
    {
        string s;
        cin >> s;
        if (s == "push")
        {
            int n;
            cin >> n;
            q.push(n);
        }
        else if (s == "pop")
        {
            if (!q.empty())
            {
                cout << q.front() << "\n";
                q.pop();
            }
            else cout << "-1\n";
        }
        else if (s == "size")
        {
            cout << q.size() << "\n";
        }
        else if (s == "empty")
        {
            cout << q.empty() << "\n";
        }
        else if (s == "front")
        {
            if (!q.empty())
            {
                cout << q.front() << "\n";
            }
            else cout << "-1\n";
        }
        else if (s == "back")
        {
            if (!q.empty())
            {
                cout << q.back() << "\n";
            }
            else cout << "-1\n";
        }
    }
}

File: 백준/Silver/18258. 큐 2/큐 2.cs
using System.Text;

namespace Baekjoon_18258
{
    internal class Program
    {
        static void Main(string[] args)
        {
            StringBuilder sb = new StringBuilder();

            Queue<int> queue = new Queue<int>();

            int N = int.Parse(Console.ReadLine());
            int back = 0;
            for (int i = 0; i < N; i++)
            {
                string[] input = Console.ReadLine().Split();
                string fun = input[0];
                switch (fun)
                {
                    case "push":
                        int value = int.Parse(input[1]);
                        back = value;
                        queue.Enqueue(value);
                        break;

                    case "pop":
                        if (queue.Count == 0)
                        {
                            sb.AppendLine("-1");
                        }
                        else
                        {
                            sb.AppendLine(queue.Dequeue().ToString());

                        }
                        break;
                    case "size":
                        sb.AppendLine(queue.Count + "");
                        break;
                    case "empty":
                        if (queue.Count == 0)
                        {
                            sb.AppendLine("1");
                        }
                        else
                        {
                            sb.AppendLine("0");
                        }
                        break;

                    case "front":
                        if (queue.Count == 0)
                        {
                            sb.AppendLine("-1");
                        }
                        else
                        {
                            sb.AppendLine(queue.Peek().ToString());
                        }
                        break;

                    case "back":
                        if (queue.Count == 0)
                        {
                            sb.AppendLine("-1");
                        }
                        else
                        {
                            sb.AppendLine(back.ToString());
                        }

                        break;
                }
            }
            Console.Write(sb.ToString());
        }
    }
}